src/lib.rs

```diff
@@ -240,3 +242,59 @@ pub struct SigElement {
     pub start: usize,
     pub end: usize,
 }
+
+#[derive(Debug, Clone, RustcDecodable, RustcEncodable)]
+pub struct Borrows {
+    pub ref_id: Id,
+    pub assignments: Vec<Assignment>,
+    pub loans: Vec<Loan>,
+    pub moves: Vec<Move>,
+}
+
+#[derive(Debug, RustcDecodable, RustcEncodable, Clone, Copy)]
+pub enum BorrowKind {
+    ImmBorrow,
+    MutBorrow,
+}
+
+#[derive(Debug, Clone, RustcDecodable, RustcEncodable)]
+pub enum CauseData {
+    ClosureCapture(SpanData),
+    AddrOf,
+    AutoRef,
+    AutoUnsafe,
+    RefBinding,
+    OverloadedOperator,
+    ClosureInvocation,
+    ForLoop,
+    MatchDiscriminant
+}
+
+#[derive(Debug, Clone, RustcDecodable, RustcEncodable)]
+pub struct Loan {
+    pub ref_id: Id,
```

Comment:
They point to the original assignment that is being borrowed from. It's like the def in goto_def, but for borrows and moves.